kernel/object-store/src/main/java/io/mosip/commons/khazana/impl/S3Adapter.java

```diff
@@ -323,16 +326,20 @@ private AmazonS3 getConnection(String bucketName) {
                     .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(url, region)).build();
             // test connection once before returning it
             connection.doesBucketExistV2(bucketName);
-            retry = 0;
+
             return connection;
 
         } catch (Exception e) {
             if (retry >= maxRetry) {
-                LOGGER.error(SESSIONID, REGISTRATIONID,"Maximum retry limit exceeded. Could not obtain connection for "+ bucketName +". Retry count :" + retry, ExceptionUtils.getStackTrace(e));
+                // reset the connection and retry count
+                retry = 0;
+                connection = null;
+                LOGGER.error(SESSIONID, REGISTRATIONID, "Maximum retry limit exceeded. Could not obtain connection for " + bucketName + ". Retry count :" + retry, ExceptionUtils.getStackTrace(e));
                 throw new ObjectStoreAdapterException(OBJECT_STORE_NOT_ACCESSIBLE.getErrorCode(), OBJECT_STORE_NOT_ACCESSIBLE.getErrorMessage(), e);
             } else {
+                connection = null;
                 retry = retry + 1;
-                LOGGER.error(SESSIONID, REGISTRATIONID,"Exception occured while obtaining connection for "+ bucketName +". Will try again. Retry count : " + retry, ExceptionUtils.getStackTrace(e));
+                LOGGER.error(SESSIONID, REGISTRATIONID, "Exception occured while obtaining connection for " + bucketName + ". Will try again. Retry count : " + retry, ExceptionUtils.getStackTrace(e));
                 getConnection(bucketName);
```

Comment:
should we not return this object to make the retry to work ?

Reply:
in case of failed connection check it will reset connection and call same method again. Every time it fails retry count will be increased by 1.  I am calling same method again if retry limit left.